### ISSUE TITLE
502 Bad Gateway

### ISSUE THREAD

**Issue by TheZ3ro** [mayoral/telegrammer#4]
I was working on TelegrammerPlugin, a bot made on top of telegrammer and I ran into this error

It shows only if I keep the bot online for a long time 

```
 (MultiJson::ParseError)s/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse': 795: unexpected token at '<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx/1.6.2</center>
</body>
</html>
'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/multi_json-1.11.1/lib/multi_json/adapters/json_common.rb:14:in `load'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/multi_json-1.11.1/lib/multi_json/adapter.rb:21:in `load'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/multi_json-1.11.1/lib/multi_json.rb:119:in `load'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/api_response.rb:9:in `initialize'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/bot.rb:237:in `new'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/bot.rb:237:in `api_request'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/bot.rb:17:in `block in get_updates'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/bot.rb:16:in `loop'
    from /home/thezero/.rvm/rubies/ruby-2.0.0-p451/lib/ruby/gems/2.0.0/gems/telegrammer-0.0.3/lib/telegrammer/bot.rb:16:in `get_updates'
    from bin/telegrammer-plugin:24:in `<main>'
```

I'm using telegrammer gem version 0.0.3,

**Comment by mayoral** [comment]
The error is not a failure in telegrammer itself. That 502 Bad Gateway response is from the Telegram API server. It seems that from time to time their servers go down. It happened to me setting a webhook and retrying a minute later the response was ok.

I'm going to add a new exception when a Bad Gateway is returned so people can catch it and do a better retry sending operations to the API.

**Comment by mayoral** [comment]
(Off-Topic) I've just created this wiki page: https://github.com/mayoral/telegrammer/wiki/People-using-Telegrammer. Can I add your project to that page?

**Comment by TheZ3ro** [comment]
Thanks, I was wondering if it was possible to handle the exception to prevent the program to end,
I didn't have time to send a pull request :smile:

**Comment by TheZ3ro** [comment]
(OT) Yes you can of course :+1:

**Comment by mayoral** [comment]
Hi again.

I've just released version 0.1.0, addressing this issue. Now, if the gem receives a 50x status code it'll raise a `ServiceUnavailableError`. Also I've fixed some data types and did a better documentation. Hope you'll enjoy it :)
